airbyte-integrations/connectors/source-shopify/source_shopify/schemas/discount_codes.json

```diff
@@ -36,6 +49,64 @@
       "description": "The type of discount applied by the discount code, such as a percentage or fixed amount off",
       "type": ["null", "string"]
     },
+    "typename": {
+      "description": "The typename of the discount",
+      "type": ["null", "string"]
+    },
+    "starts_at": {
+      "description": "The date and time when the discount code is activated.",
+      "type": ["null", "string"],
+      "format": "date-time"
+    },
+    "ends_at": {
+      "description": "The date and time when the discount code is deactivated.",
+      "type": ["null", "string"],
+      "format": "date-time"
+    },
+    "status": {
+      "description": "The status of the Discount",
+      "type": ["null", "string"]
+    },
+    "title": {
+      "description": "The title of the discount.",
+      "type": ["null", "string"]
+    },
+    "usage_limit": {
+      "description": "The maximum number of times that the discount can be used.",
+      "type": ["null", "integer"]
+    },
+    "applies_once_per_customer": {
+      "description": "Whether the discount can be applied only once per customer.",
+      "type": ["null", "boolean"]
+    },
+    "async_usage_count": {
+      "description": "The number of times that the discount has been used.",
+      "type": ["null", "integer"]
+    },
+    "codes_count": {
+      "description": "The number of redeem codes for the discount.",
+      "type": ["null", "object"],
+      "additionalProperties": true,
+      "properties": {
+        "count": {
+          "description": "The count of elements.",
+          "type": ["null", "integer"]
+        }
+      }
+    },
+    "total_sales": {
+      "description": "The total sales from orders where the discount was used.",
+      "type": ["null", "object"],
+      "additionalProperties": true,
```

Comment:
It's something from the "best practices", but you're right, there is no need to have it since we have a strict field selection for query and the schema is build in accordance to that as well - we don't need to have "additionalProperties" here. It's neither good nor bad to leave it there.